Bug with angular-inviewport v4.0.0

Hi,

I have an error after updating your package thisissoon/angular-inviewport to v4.0.0 :

> ERROR in node_modules/@thisissoon/angular-scroll-collapse/app/scroll-collapse/scroll-collapse.directive.d.ts(2,10): error TS2305: Module '".../node_modules/@thisissoon/angular-inviewport/thisissoon-angular-inviewport"' has no exported member 'WindowRef'.

Is it possible to fix it ?

Thank you

--- a/package.json
+++ b/package.json
@@ -51,7 +51,7 @@
   },
   "peerDependencies": {
     "@angular/core": ">=5.0.0 <7.0.0",
-    "@thisissoon/angular-inviewport": ">=3.0.0 <4.0.0",
+    "@thisissoon/angular-inviewport": ">=4.0.0 <5.0.0",
     "rxjs": ">=6.0.0 <7.0.0"
   },
   "devDependencies": {
@@ -68,7 +68,7 @@
     "@angular/platform-browser": "^6.1.9",
     "@angular/platform-browser-dynamic": "^6.1.9",
     "@angular/router": "^6.1.9",
-    "@thisissoon/angular-inviewport": "^3.0.1",
+    "@thisissoon/angular-inviewport": "^4.0.0",
     "@thisissoon/schematics": "^1.0.0-alpha.18",
     "@types/jasmine": "~2.8.3",
     "@types/jasminewd2": "~2.0.2",

--- a/src/app/app.module.ts
+++ b/src/app/app.module.ts
@@ -1,18 +1,12 @@
 import { NgModule } from '@angular/core';
 import { BrowserModule } from '@angular/platform-browser';
-import { InViewportModule, WindowRef } from '@thisissoon/angular-inviewport';
+import { InViewportModule } from '@thisissoon/angular-inviewport';
 
 import { ScrollCollapseModule } from './scroll-collapse';
 import { AppComponent } from './app.component';
 
 @NgModule({
-  imports: [
-    BrowserModule,
-    InViewportModule.forRoot([
-      { provide: WindowRef, useFactory: () => window },
-    ]),
-    ScrollCollapseModule,
-  ],
+  imports: [BrowserModule, InViewportModule, ScrollCollapseModule],
   declarations: [AppComponent],
   bootstrap: [AppComponent],
 })

--- a/src/app/scroll-collapse/scroll-collapse.directive.spec.ts
+++ b/src/app/scroll-collapse/scroll-collapse.directive.spec.ts
@@ -1,6 +1,5 @@
 import { fakeAsync, tick } from '@angular/core/testing';
 import { ElementRef, SimpleChanges } from '@angular/core';
-import { WindowRef } from '@thisissoon/angular-inviewport';
 
 import { ScrollCollapseDirective } from './scroll-collapse.directive';
 import { Direction } from './shared/direction.enum';

--- a/src/app/scroll-collapse/scroll-collapse.directive.ts
+++ b/src/app/scroll-collapse/scroll-collapse.directive.ts
@@ -8,6 +8,7 @@ import {
   OnDestroy,
   Output,
   EventEmitter,
+  Inject,
 } from '@angular/core';
 import { fromEvent, merge, Subject } from 'rxjs';
 import {
@@ -19,7 +20,7 @@ import {
   distinctUntilChanged,
   tap,
 } from 'rxjs/operators';
-import { WindowRef } from '@thisissoon/angular-inviewport';
+import { WINDOW } from '@thisissoon/angular-inviewport';
 
 import { Viewport, Direction } from './shared';
 import * as eventData from './shared/event-data';
@@ -153,7 +154,7 @@ export class ScrollCollapseDirective implements AfterViewInit, OnDestroy {
   constructor(
     private el: ElementRef,
     private ngZone: NgZone,
-    private windowRef: WindowRef,
+    @Inject(WINDOW) private window: Window,
   ) {}
   /**
    * Subscribe to window resize events as an observable
@@ -167,14 +168,14 @@ export class ScrollCollapseDirective implements AfterViewInit, OnDestroy {
     // in an platform without the DOM
     if (typeof el.getBoundingClientRect === 'function') {
       const elBounds = el.getBoundingClientRect();
-      this.originalTop = elBounds.top + this.windowRef.pageYOffset;
+      this.originalTop = elBounds.top + this.window.pageYOffset;
     }
     this.originalHeight = el.offsetHeight;
 
     this.ngZone.runOutsideAngular(() => {
       merge(
-        fromEvent(this.windowRef as any, eventData.eventScroll),
-        fromEvent(this.windowRef as any, eventData.eventResize),
+        fromEvent(this.window as any, eventData.eventScroll),
+        fromEvent(this.window as any, eventData.eventResize),
       )
         .pipe(
           startWith(null),
@@ -261,10 +262,10 @@ export class ScrollCollapseDirective implements AfterViewInit, OnDestroy {
    */
   public getViewport(): Viewport {
     return {
-      height: this.windowRef.innerHeight,
-      width: this.windowRef.innerWidth,
-      pageYOffset: this.windowRef.pageYOffset,
-      pageXOffset: this.windowRef.pageXOffset,
+      height: this.window.innerHeight,
+      width: this.window.innerWidth,
+      pageYOffset: this.window.pageYOffset,
+      pageXOffset: this.window.pageXOffset,
     };
   }
   /**

